### PR TITLE
Idva6 1142 fetch members for acsp number

### PR DESCRIPTION
**Pull request by krishna-patel-ch**
__Short description outlining key changes/additions__

New endpoint to fetch users that are members of a given ACSP.

__JIRA Ticket Number__

IDVA6-1142

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [x] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/service/AcspMembersService.java [hunk 1]
```diff
@@ -44,4 +53,27 @@ public List<AcspMembership> fetchAcspMemberships(final User user, final boolean
 
     return acspMembershipListMapper.daoToDto(acspMembers, user);
   }
+
+    @Transactional( readOnly = true )
+    public AcspMembers fetchAcspMembers( final AcspDataDao acspData, final boolean includeRemoved, final String userId, final String role, final int pageIndex, final int itemsPerPage ) {
+        final var acspNumber = acspData.getId();
+
+        if ( Objects.isNull( acspNumber ) ){
+            throw new NullPointerException( "acspNumber is null." );
```
Comment (tferns-ch) [inline src/main/java/uk/gov/companieshouse/acsp/manage/users/service/AcspMembersService.java:62]: Illegal argument is probably more appropriate, as a null pointer hasn't actually happened yet, and it's more reflective of a missing required thing:

```suggestion
            throw new IllegalArgumentException( "acspNumber is null." );
```

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoConfig.java [hunk 2]
```diff
@@ -16,6 +18,11 @@
 @EnableMongoAuditing(dateTimeProviderRef = "mongodbDatetimeProvider")
 public class MongoConfig {
 
+    @Bean
+    public MongoCustomConversions mongoCustomConversions() {
+        return new MongoCustomConversions(List.of(new MongoRoleReadingConverter(), new MongoRoleWritingConverter()));
+    }
+
     @Bean
```
Comment (tferns-ch) [inline src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoConfig.java:26]: When exactly are these being used?

**Review by tferns-ch — APPROVED** [review]
Solid work.  Solid test coverage, can't really find much to criticize, well done!  I've mentioned one minor thing and just a question too.  Could you maybe make sure you're using the Google style formatter plugin?  Looks like the code has 4 spaces of indentation and lots of other differences from the style guide: https://google.github.io/styleguide/javaguide.html#

**Comment by tferns-ch** [comment]
Could you also maybe resolve the minor code smells? https://code-analysis.platform.aws.chdev.org/project/issues?id=uk.gov.companieshouse%3Aacsp-manage-users-api&pullRequest=21&resolved=false&types=CODE_SMELL

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoRoleReadingConverter.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.companieshouse.acsp.manage.users.configuration;
```
Comment (tferns-ch) [inline src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoRoleReadingConverter.java:1]: Unit tests are probably easy to make for these too

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/service/AcspMembersService.java (hunk 1 above)
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/acsp/manage/users/service/AcspMembersService.java:62]: Changed it to IllegalArgumentException

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoConfig.java (hunk 2 above)
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoConfig.java:26]: Whenever you query the database e.g. @Query annotation, it applies this automatically under the hood.

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoRoleReadingConverter.java (hunk 3 above)
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoRoleReadingConverter.java:1]: Added tests

Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoRoleReadingConverter.java:1]: Resolved code smells

File: src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoConfig.java (hunk 2 above)
Comment (tferns-ch) [reply src/main/java/uk/gov/companieshouse/acsp/manage/users/configuration/MongoConfig.java:26]: Oh that's cool.  Thanks

**Review by tferns-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Pull request merged**